node/src/events.rs

```diff
@@ -219,6 +228,37 @@ pub enum NodeEvent {
         /// A human readable error.
         error: String,
     },
+
+    /// A new head that was observed by the node.
+    NewSubjectiveHead {
+        /// The height of the head.
+        height: u64,
+    },
+
+    /// A new header was added from HeaderSub.
+    AddedHeaderFromHeaderSub {
+        /// The height of the header.
+        height: u64,
+    },
+
+    /// Fetching header of network head in progress.
+    FetchingHeadHeader,
+
+    /// Fetching headers for a specific block range.
+    FetchingHeaders {
```

Comment:
I think we should also have an event when we succeeded fetching and inserted them

Reply:
Done